src/main/java/liquibase/ext/spanner/CloudSpanner.java

```diff
@@ -265,21 +269,20 @@ public String escapeStringForDatabase(String string) {
   }
 
   @Override
-  public Dialect getDialect() throws SQLException {
-    DatabaseConnection conn = getConnection();
+  public Dialect getDialect() {
+    try {
+      DatabaseConnection conn = getConnection();
 
-    if (conn instanceof JdbcConnection) {
-      Connection underlying = ((JdbcConnection) conn).getUnderlyingConnection();
-      if (underlying.isWrapperFor(CloudSpannerJdbcConnection.class)) {
-        return underlying.unwrap(CloudSpannerJdbcConnection.class).getDialect();
+      if (conn instanceof JdbcConnection) {
+        Connection underlying = ((JdbcConnection) conn).getUnderlyingConnection();
+        if (underlying.isWrapperFor(CloudSpannerJdbcConnection.class)) {
+          return underlying.unwrap(CloudSpannerJdbcConnection.class).getDialect();
+        }
       }
+    } catch (SQLException e) {
+      throw new RuntimeException("Failed to get dialect from connection", e);
     }
-    return null;
-  }
 
-  // TODO: Temporarily solution. Will need remove after fix the TABLE_CATALOG bug in the emulator
```

Comment:
(no need to fix here, just to let you know): The bug in the emulator has been fixed. The latest version contains the correct value for TABLE_CATALOG in PG.